pkg/api/steve/projectresources/handlers.go

```diff
@@ -0,0 +1,853 @@
+		"kind":         "APIResourceList",
+		"apiVersion":   "v1",
+		"groupVersion": groupVersion,
+		"resources":    h.apis.List(),
+	}
+	apiResourceBytes, err := json.Marshal(resp)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Write(apiResourceBytes)
+}
+
+// forwarder handles global resource requests that have no particular project or namespace specified.
+// In this case, there is nothing special about the request so the regular client can handle it without any preprocessing.
+func (h *handler) forwarder(w http.ResponseWriter, r *http.Request) {
+	logrus.Tracef("[%s] handling request %s", apiServiceName, r.URL.Path)
+	vars := mux.Vars(r)
+	resource, err := h.gvrFromVars(vars)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	resourceClient := h.dynamicClient.Resource(resource)
+	if !h.hasAccess(r, "", resource) {
+		h.returnEmpty(r.Context(), w, resource, resourceClient)
+		return
+	}
+	opts := metav1.ListOptions{}
+	paramCodec.DecodeParameters(r.URL.Query(), metav1.SchemeGroupVersion, &opts)
+	resources, err := resourceClient.List(r.Context(), opts)
+	if apierrors.IsNotFound(err) {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	returnResp(w, resources)
+}
+
+// unscopedHandler handles requests for resources in namespaces that don't belong to any project.
+func (h *handler) unscopedHandler(w http.ResponseWriter, r *http.Request) {
+	logrus.Tracef("[%s] handling request %s", apiServiceName, r.URL.Path)
+	vars := mux.Vars(r)
+	op := vars["op"]
+	var projectsOrNamespaces []string
+	if value, ok := vars[projectsOrNamespacesVar]; ok {
+		projectsOrNamespaces = strings.Split(value, ",")
+	}
+	filteredNamespaces := make(map[string]struct{})
+	resource, err := h.gvrFromVars(vars)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	resourceClient := h.dynamicClient.Resource(resource)
+	for _, n := range projectsOrNamespaces {
+		_, err := h.namespaceCache.Get(n)
+		if err != nil && !apierrors.IsNotFound(err) {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		filteredNamespaces[n] = struct{}{}
+	}
+	// get namespaces that don't have the projectID label
+	nss, err := h.namespaceCache.List(orphanNamespaceSelector)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	finalNamespaces := make(map[string]*corev1.Namespace)
+	for _, ns := range nss {
+		// no namespaces in the selector so all can be included
+		if len(projectsOrNamespaces) == 0 {
+			finalNamespaces[ns.Name] = ns
+			continue
+		}
+		// only include resources in namespaces specified in the selector, or NOT specified in the selector if the operator is !
+		_, ok := filteredNamespaces[ns.Name]
+		if (op != notOp && ok) || (op == notOp && !ok) {
+			finalNamespaces[ns.Name] = ns
+		}
```

Comment:
Optionally, you can use another label selector above to filter out these resources so that you don't have to do this here. Something like the below should do the trick:

```golang
// orphanRequirement, in this example, is a requirement like this:
// labels.NewRequirement(nslabels.ProjectIDFieldLabel, selection.DoesNotExist, nil)
namespaceRequirement, err := labels.NewRequirement(corev1.LabelMetadataName, selection.NotIn, nsNames)
if err != nil {
	http.Error(w, err.Error(), http.StatusInternalServerError)
	return
}
selector = labels.NewSelector().Add(*orphanRequirement, *namespaceRequirement)

```

This would allow you to just list and convert, without needing to filter later.